snapshots/manager.go

```diff
@@ -376,3 +411,50 @@ func IsFormatSupported(snapshotter types.ExtensionSnapshotter, format uint32) bo
 	}
 	return false
 }
+
+// SnapshotIfApplicable takes a snapshot of the current state if we are on a snapshot height.
+// It also prunes any old snapshots. The snapshotting and pruning happen in separate goroutines.
+func (m *Manager) SnapshotIfApplicable(height int64) {
+	if m == nil {
+		return
+	}
+	if !m.shouldTakeSnapshot(height) {
+		m.logger.Debug("snapshot is skipped", "height", height)
+		return
+	}
+	go m.snapshot(height)
```

Comment:
```suggestion
	m.snapshot(height)
```
The caller can `go m.SnapshotIfApplicable` if they so choose.

Reply:
I'll do the change but my concern is that then we would be unnecessarily spinning up new goroutines just to do validations that can be done synchronously

Reply:
This code currently (and unavoidably) spins up a goroutine to do the snapshotting. What's the concern about doing everything synchronously?

Reply:
You are right. I had the impression that there was a blocking operation in `m.snapshot(...)`